### PR TITLE
Add's checks for the Kubectl Kubernetes commands

### PR DESCRIPTION
**Pull request by mcmarkj**
This PR adds checks for:

📶  The version of Kubernetes installed via the `kubectl version` command - so you can check the version installed by doing:
```
KubectlInstalled(minimum_version=1.18)
```
This means that anything in the 1.18 or above is acceptable.

🔑  Checks that contexts are configured in the default kubectl config file. You also can optionally pass in a provision_command to inform users how to configure the context (i.e. by using gcloud commands / aws etc.) 
```
    KubectlContextExists(context_name="gke_my-example-project_europe-west2-a_my-cluster")
```

```
    KubectlContextExists(context_name="gke_my-example-project_europe-west2-a_my-cluster" provision_command="gcloud container clusters get-credentials my-cluster --zone europe-west2-a --project my-example-project")
```

### CODE REVIEW

File: daktari/checks/kubernetes.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class KubectlContextExists(Check):
+    def __init__(self, context_name: str, provision_command: str = None):
+        self.context_name = context_name
+        self.name = f"kubectl.contextExists.{context_name}"
+
+        if provision_command is None:
+            self.suggestions = {
+                OS.GENERIC: f"""A kubectl context is missing, :
+                               <cmd>{provision_command}</cmd>"""
+            }
```
Comment (alexburlton-sonocent) [inline daktari/checks/kubernetes.py:57]: this looks a bit weird to me? Wondering if we can do something as simple as:

```
self.suggestions = {
    OS.GENERIC: provision_command
}
```

Then we'll still always print "dev_context_blah is not configured for the current user", but with the suggestion if one has been given?

File: daktari/checks/kubernetes.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            OS.GENERIC: "Install kubectl: https://kubernetes.io/docs/tasks/tools/#kubectl",
+        }
+
+    def check(self) -> CheckResult:
+        installed_version = get_kubectl_version()
+        if installed_version is None:
+            return self.failed("Kubectl is not installed")
+
+        if self.minimum_version is None:
+            return self.passed("Kubectl is installed")
```
[the remark this answers is not in the document]
Comment (mcmarkj) [reply daktari/checks/kubernetes.py:26]: Sound logical to me!

File: daktari/checks/kubernetes.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            installed_version >= self.minimum_version,
+            f"Kubectl version is <not/> >={self.minimum_version} ({installed_version})",
+        )
+
+
+version_pattern = re.compile("Client Version: v([0-9]+.[0-9]+)")
+
+
+def get_kubectl_version() -> Optional[float]:
+    raw_version = get_stdout("kubectl version --client=true --short")
```
[the remark this answers is not in the document]
Comment (mcmarkj) [reply daktari/checks/kubernetes.py:38]: 😅  If you don't you get a crazy output around the server versions, go versions and lots of nonsense we probs don't care about.

File: daktari/checks/kubernetes.py (hunk 1 above)
Comment (mcmarkj) [reply daktari/checks/kubernetes.py]: Coolio - makes perfect sense.

File: daktari/checks/kubernetes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexburlton-sonocent) [reply daktari/checks/kubernetes.py:38]: yeah I had a play about locally and saw that. Just a bit :exploding_head: that `kubectl` would implement it so cumbersomely!

**Review by alexburlton-sonocent — APPROVED** [review]

**Pull request merged**